File: pset6/vigenere.py
from cs50 import get_string
import sys


if(len(sys.argv) != 2):
    print("Usage: python file key")
    exit(1)
i = 0
# get the key
key = str(sys.argv[1])

if key.isalpha() != True:
    exit(1)

# get the plain text
print("Enter the text: ", end="")
plain_text = get_string()

ciphertext = ""
for c in plain_text:
    if(c.isalpha()):
        if(c.islower()):
            ciphertext += chr(((ord(c) - 97 + (ord(key[i % len(key)].lower()) - 97)) % 26) + 97)
        else:
            ciphertext += chr(((ord(c) - 65 + (ord(key[i % len(key)].lower()) - 97)) % 26) + 65)
        i += 1
    else:
        ciphertext += c
# cyphertext = "A"
print("ciphertext:", ciphertext)

File: pset2/vigenere.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
int main(int argc, string argv[]){
    string k = argv[1];
    int j = 0;
    printf("plaintext: ");
    string str = get_string();
    int s = strlen(str);
    char arr[s];
    for(int i = 0; i < strlen(str); i++){
        arr[i] = str[i];
    }
    printf("ciphertext: ");
    for(int i = 0; i < strlen(str); i++){
        if(arr[i] >64 && arr[i] < 91){

            if(arr[i] + toupper(k[j%strlen(k)]) -65 < 91)
            printf("%c", arr[i] + toupper(k[j%strlen(k)])-65);
            else{
                int a = 90 - arr[i];
                printf("%c", 64 + toupper(k[j%strlen(k)])-a-65);
            }
            j++;
        } else if(arr[i] > 96 && arr[i] < 123){

            if(arr[i] + tolower(k[j%strlen(k)])-97 < 123)
            printf("%c", arr[i] + tolower(k[j%strlen(k)])-97);
            else{
                int a = 123 - arr[i];
                printf("%c", 97 + tolower(k[j%strlen(k)])-a-97);
            }
            j++;
        } else
            printf("%c", arr[i]);
    }
    printf("\n");
    (void)argc;
}